frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js

```diff
@@ -241,39 +482,69 @@ export default function InteractiveWaveformModal({
     }
   }, []);
 
+  // Update baseData anytime the remove dups checkbox state changes, except when set during load of existing changes
   useEffect(() => {
+    if (loadStatus === LOAD_STATUSES.LOADING_EXISTING) {
+      return;
+    }
     if (removeDupsChecked) {
-      const currentPeaksValleysCopy = JSON.parse(JSON.stringify(editablePeaksValleys));
-      // remove duplicate peaks and valleys per well
-      for (const well in currentPeaksValleysCopy) {
-        currentPeaksValleysCopy[well] = removeWellSpecificDuplicates(well);
+      const baseDataCopy = deepCopy(baseData);
+      for (const well in baseDataCopy) {
+        const { duplicateFeatureIndices } = customAnalysisSettings[well];
+        baseDataCopy[well][0] = removeDups(baseDataCopy[well][0], duplicateFeatureIndices.peaks);
+        baseDataCopy[well][1] = removeDups(baseDataCopy[well][1], duplicateFeatureIndices.valleys);
       }
-      setBaseData(currentPeaksValleysCopy);
-    } else {
+      setBaseData(baseDataCopy);
+    } else if (originalAnalysisData.featuresForWells) {
       // if unchecked, revert back to previous peaks and valleys
-      setBaseData(originalData.peaksValleys);
+      setBaseData(originalAnalysisData.featuresForWells);
     }
   }, [removeDupsChecked]);
 
+  // Update the custom peaks and valleys correctly after baseData is updated.
+  // Currently baseData will only be updated when it is initially set for all wells, when the remove dups checkbox state changes (in the useEffect above),
+  // and when loading existing data.
   useEffect(() => {
-    // This is just to update the editable peaks and valleys correctly after the useEffect above
-    setEditablePeaksValleys(baseData);
+    if (loadStatus === LOAD_STATUSES.LOADING_EXISTING) {
+      return;
+    }
+    for (const well of wellNames) {
+      if (baseData[well]) {
+        customAnalysisSettingsInitializers.featureIndices(well, "peaks", baseData[well][0]);
+        customAnalysisSettingsInitializers.featureIndices(well, "valleys", baseData[well][1]);
+      }
+    }
   }, [baseData]);
 
+  // A spinner is displayed until the waveform data for the currently selected well is loaded.
+  // This switches from the spinner to the WaveformGraph once that happens, as well as setting the default
+  // threshold lines if they aren't already set (i.e. loaded from sessionStorage)
   useEffect(() => {
-    // updates changelog when peaks/valleys and start/end times change
-    if (undoing) {
-      setUndoing(false);
-    } else {
-      updateChangelog();
+    switch (loadStatus) {
+      case LOAD_STATUSES.LOADING_NEW: {
+        for (const well of wellNames) {
+          setBothLinesToDefault(well);
+        }
+        // fall through expected here
```

Comment:
what do you mean by this?

Reply:
Added a comment to indicate that there should be no `break`, i.e. this case should fall through to the next